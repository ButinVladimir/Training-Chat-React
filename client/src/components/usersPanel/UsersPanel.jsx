import React from 'react';
import PropTypes from 'prop-types';
import User from './user/UserContainer';
import './style.css';

function UsersPanel({ usersList }) {
  const convertedUsersList = usersList.map(user => (
    <User
      key={user}
      user={user}
    />
  ));

  return (
    <div className="chat-users-list-container">
      <ul className="users-list">
        {convertedUsersList}
      </ul>
    </div>
  );
}

UsersPanel.propTypes = {
  usersList: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
};

export default UsersPanel;
