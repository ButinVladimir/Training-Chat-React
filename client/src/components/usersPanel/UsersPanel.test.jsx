import React from 'react';
import { shallow } from 'enzyme';
import UsersPanel from './UsersPanel';

describe('UsersPanel', () => {
  it('renders without crashing', () => {
    const component = shallow(<UsersPanel usersList={['user 1', 'user 2']} onClickUser={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});
