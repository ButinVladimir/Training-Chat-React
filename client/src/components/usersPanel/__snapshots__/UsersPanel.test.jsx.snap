// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersPanel renders without crashing 1`] = `
<div
  className="chat-users-list-container"
>
  <ul
    className="users-list"
  >
    <Connect(User)
      key="user 1"
      user="user 1"
    />
    <Connect(User)
      key="user 2"
      user="user 2"
    />
  </ul>
</div>
`;
